pkg/config/config.go

```diff
@@ -523,6 +524,113 @@ type Pipeline struct {
 	Environment map[string]string `json:"environment,omitempty" yaml:"environment,omitempty"`
 }
 
+// getGitSBOMPackage creates an SBOM package for Git based repositories.
+// Returns nil package and nil error if the repository is not from a supported platform or
+// if neither a tag of expectedCommit is not provided
+func getGitSBOMPackage(repo, tag, expectedCommit string, idComponents []string, licenseDeclared string, hint string) (*sbom.Package, error) {
+	var repoType, namespace, name, ref string
+	var downloadLocation string
+
+	repoURL, err := url.Parse(repo)
+	if err != nil {
+		return nil, err
+	}
+
+	if expectedCommit != "" {
+		ref = expectedCommit
+	} else if tag != "" {
+		ref = tag
+	} else {
+		// No commit or tag provided - we're done
+		return nil, nil
+	}
+
+	trimmedPath := strings.TrimPrefix(repoURL.Path, "/")
+	namespace, name, _ = strings.Cut(trimmedPath, "/")
+	name = strings.TrimSuffix(name, ".git")
+
+	switch {
+	case repoURL.Host == "github.com":
+		repoType = purl.TypeGithub
+		downloadLocation = fmt.Sprintf("%s://github.com/%s/%s/archive/%s.tar.gz", repoURL.Scheme, namespace, name, ref)
+
+	case repoURL.Host == "gitlab.com":
+		repoType = purl.TypeGitlab
+		downloadLocation = fmt.Sprintf("%s://gitlab.com/%s/%s/-/archive/%s/%s.tar.gz", repoURL.Scheme, namespace, name, ref, ref)
+
+	case strings.HasPrefix(repoURL.Host, "gitlab") || hint == "gitlab":
+		repoType = purl.TypeGeneric
+		downloadLocation = fmt.Sprintf("%s://%s/%s/%s/-/archive/%s/%s.tar.gz", repoURL.Scheme, repoURL.Host, namespace, name, ref, ref)
+
+	default:
+		repoType = purl.TypeGeneric
+		namespace = ""
```

Comment:
@javacruft Is there a reason why this field is explicitly set to empty?
Its causing an issue where supplier is not correctly set: https://github.com/chainguard-dev/customer-issues/issues/2375

Reply:
we can set the namespace to the git repo if that is okay

Reply:
I think we should revert to the previous behaviour and use the supplier - let me raise a PR to fix.

Reply:
https://github.com/chainguard-dev/melange/pull/2081